web/meetup-facilitator/src/App.jsx

```diff
@@ -1,34 +1,54 @@
-import { useState } from 'react'
-import reactLogo from './assets/react.svg'
-import viteLogo from '/vite.svg'
 import './App.css'
+import Button from './components/Button'
+import { useRef, useState } from 'react'
 
 function App() {
-  const [count, setCount] = useState(0)
+
+  const buttonStyles = 'bg-green-700'
+  const groupName = useRef('')
+  const [isValid, setIsValid] = useState(true);
+
+
+  const updateGroupName = (e) => {
+    groupName.current = e.target.value;
+    console.log(groupName.current);
+  }
+
+  const onChange = (e) => {
+    updateGroupName(e);
+    validateInput();
+  }
+
+  const onSubmit = (e) => {
+    e.preventDefault();
+    console.log("submitting " + groupName.current);
+
+    // Validate once on submit
+    validateInput();
+
+    // if isValid -> make API call
+
+  }
+
+  const validateInput = () => {
+    console.log('validating input: ' + groupName.current);
+
+    if (groupName.current === '')
```

Comment:
In the database, we're making the group names have a maximum length of 30 characters. Could you implement that check here as well?

Reply:
Will fix in later PR